syntax = "proto3";

import "eigen.proto";

message UnivariateMixtureAtom {
  double mean = 1;
  double stdev = 2;
}

message UnivariateMixtureState {
    int32 num_components = 1;
    repeated double weights = 2 [packed=true];
    repeated UnivariateMixtureAtom atoms = 3;
}

message UnivariateState {
  message GroupParams {
    repeated double weights = 1 [packed=true];
    repeated int32 cluster_allocs = 2 [packed=true];
  }

  int32 num_components = 1;

  repeated UnivariateMixtureAtom atoms = 2;

  repeated GroupParams groupParams = 3;

  double rho = 4;

  EigenMatrix Sigma = 5;

  repeated double regression_coefficients = 6 [packed=true];
}


message HdpState {
  message GroupParams {
    repeated int32 cluster_size = 1 [packed=true];
    repeated int32 cluster_allocs = 2 [packed=true];
  }

  int32 num_components = 1;
  repeated UnivariateMixtureAtom atoms = 2;
  repeated GroupParams groupParams = 3;
  repeated double betas = 4 [packed=true];

  message HyperParams {
    double mu0 = 1;
    double a = 2;
    double b = 3;
    double lamb = 4;
  }

  HyperParams hyper_params = 5;
  double alpha = 6;
  double gamma = 7;
}
